src/uct/sm/scopy/knem/knem_iface.c

```diff
@@ -25,73 +25,47 @@ static ucs_status_t uct_knem_iface_query(uct_iface_h tl_iface,
 {
     uct_knem_iface_t *iface = ucs_derived_of(tl_iface, uct_knem_iface_t);
 
-    uct_base_iface_query(&iface->super.super, iface_attr);
-
-    /* default values for all shared memory transports */
-    iface_attr->cap.put.min_zcopy       = 0;
-    iface_attr->cap.put.max_zcopy       = SIZE_MAX;
-    iface_attr->cap.put.opt_zcopy_align = 1;
-    iface_attr->cap.put.align_mtu       = iface_attr->cap.put.opt_zcopy_align;
-    iface_attr->cap.put.max_iov         = uct_sm_get_max_iov();
-
-    iface_attr->cap.get.min_zcopy       = 0;
-    iface_attr->cap.get.max_zcopy       = SIZE_MAX;
-    iface_attr->cap.get.opt_zcopy_align = 1;
-    iface_attr->cap.get.align_mtu       = iface_attr->cap.get.opt_zcopy_align;
-    iface_attr->cap.get.max_iov         = uct_sm_get_max_iov();
-
-    iface_attr->cap.am.max_iov         = 1;
-    iface_attr->cap.am.opt_zcopy_align = 1;
-    iface_attr->cap.am.align_mtu       = iface_attr->cap.am.opt_zcopy_align;
-
-    iface_attr->iface_addr_len         = 0;
-    iface_attr->device_addr_len        = uct_sm_iface_get_device_addr_len();
-    iface_attr->ep_addr_len            = 0;
-    iface_attr->max_conn_priv          = 0;
-    iface_attr->cap.flags              = UCT_IFACE_FLAG_GET_ZCOPY |
-                                         UCT_IFACE_FLAG_PUT_ZCOPY |
-                                         UCT_IFACE_FLAG_PENDING   |
-                                         UCT_IFACE_FLAG_CONNECT_TO_IFACE;
-    iface_attr->latency.overhead       = 80e-9; /* 80 ns */
-    iface_attr->latency.growth         = 0;
-    iface_attr->bandwidth.shared       = iface->super.config.bandwidth;
-    iface_attr->bandwidth.dedicated    = 0;
-    iface_attr->overhead               = 0.25e-6; /* 0.25 us */
+    uct_scopy_iface_query(&iface->super, iface_attr);
+
+    iface_attr->bandwidth.shared    = iface->super.super.config.bandwidth;
+    iface_attr->bandwidth.dedicated = 0;
+    iface_attr->overhead            = 0.25e-6; /* 0.25 us */
```

Comment:
would need to set `iface_addr_len` to 0

Reply:
I'm going to fix it along with other comments

Reply:
done